test/utils_tests.cc

```diff
@@ -0,0 +1,36 @@
+
+#include <catch2/catch.hpp>
+
+#include <celerity.h>
+
+#include "affinity.h"
+#include "test_utils.h"
+
+namespace celerity {
+namespace detail {
+	namespace utils {
+
+		TEST_CASE("affinity check", "[utils]") {
+			// REQUIRE_FALSE(runtime::is_initialized());
+
+#ifdef _WIN32
+			SECTION("in Windows") {
+				DWORD_PTR cpu_mask = 1;
+				SetProcessAffinityMask(GetCurrentProcess(), cpu_mask);
+			}
+#else
+#include <sched.h>
```

Comment:
Never ever include headers inside a namespace, unless you are absolutely sure that is what you want. This pollutes the `celerity::detail::utils` namespace with whatever is inside `sched.h`, which is very likely not what you want.